File: 206-reverse-linked-list/reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        prev = None
        pres = head
        if head is not None:nxt = head.next
        else: return head

        while nxt is not None:
            pres.next = prev
            prev = pres
            pres = nxt
            nxt = nxt.next

        pres.next = prev
        return pres

File: 206-reverse-linked-list/reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode prev = null;
        ListNode pres = head;
        ListNode nxt;
        if (head != null) {
            nxt = head.next;
            }
        else return head;

        while (nxt != null){
            pres.next = prev;
            prev = pres;
            pres = nxt;
            nxt = nxt.next;
        }
        pres.next = prev;
        return pres; 
    }
}